operators/src/processing/time_shift.rs

```diff
@@ -316,12 +410,30 @@ where
         query: RasterQueryRectangle,
         ctx: &'a dyn QueryContext,
     ) -> Result<BoxStream<'a, Result<RasterTile2D<Self::RasterType>>>> {
+        let time_start_difference = query.time_interval.start() - self.time_interval.start();
+        let time_end_difference = query.time_interval.end() - self.time_interval.end();
+
         let query = RasterQueryRectangle {
             spatial_bounds: query.spatial_bounds,
             time_interval: self.time_interval,
             spatial_resolution: query.spatial_resolution,
         };
-        self.processor.raster_query(query, ctx).await
+
+        let stream = self.processor.raster_query(query, ctx).await?;
+
+        let stream = stream.map(move |raster| {
+            // reverse time shift for results
+            let mut raster = raster?;
+
+            raster.time = TimeInterval::new(
+                raster.time.start() + time_start_difference.num_milliseconds(),
```

Comment:
again: not sure if this is the correct semantics

Reply:
What would be correct?
The result is shifted the same amount as the query but in opposite directions.